mypy/plugins/enums.py

```diff
@@ -78,6 +78,22 @@ class SomeEnum:
     """
     enum_field_name = _extract_underlying_field_name(ctx.type)
     if enum_field_name is None:
+        # We do not know the enum field name (perhaps it was passed to a function and we only
+        # know that it _is_ a member).  All is not lost however, if we can prove that the all
+        # of the enum members have the same value-type, then it doesn't matter which member
+        # was passed in.  The value-type is still known.
+        if isinstance(ctx.type, Instance):
+            info = ctx.type.type
```

Comment:
Should we memoize this bit?  It'd probably be easy enough to pull this into a separate function an `@functools.lru_cache` it...

Reply:
I wouldn't bother. It would be quite the program where you could show a measurable performance improvement for type checking by memoizing this.